gravitee-apim-rest-api/gravitee-apim-rest-api-management-v2/gravitee-apim-rest-api-management-v2-rest/src/main/java/io/gravitee/rest/api/management/v2/rest/mapper/PageMapper.java

```diff
@@ -169,4 +173,142 @@ default JsonNode deserializeJsonConfiguration(Object configuration) throws JsonP
         }
         return null;
     }
+
+    /**
+     * Deserialize configuration and mask sensitive data using field names.
+     * This method masks known sensitive field names like privateToken, password, etc.
+     * This is a simple and reliable approach that doesn't require custom annotations.
+     */
+    @Named("deserializeConfigurationWithSensitiveDataMasking")
+    default Object deserializeConfigurationWithSensitiveDataMasking(String configuration) {
+        if (Objects.isNull(configuration)) {
+            return null;
+        }
+
+        try {
+            // Try to parse as LinkedHashMap first
+            Object config = mapper.readValue(configuration, LinkedHashMap.class);
```

Comment:
`readValue` is Generic so it directly maps to a LinkedHashMap.
Therefore the next check is useless 

This should work


```suggestion
            LinkedHashMap<String, Object> config = mapper.readValue(configuration, LinkedHashMap.class);
```